gadgets/trace_tcpconnect/program.bpf.c

```diff
@@ -64,15 +69,26 @@ const volatile __u64 targ_min_latency_ns = 0;
 #define AF_INET6 10
 
 // sockets_per_process keeps track of the sockets between:
-// - kprobe enter_tcp_connect
-// - kretprobe exit_tcp_connect
+// - kprobe inet_stream_connect
+// - kretprobe inet_stream_connect
 struct {
 	__uint(type, BPF_MAP_TYPE_HASH);
 	__uint(max_entries, MAX_ENTRIES);
 	__type(key, u32); // tid
 	__type(value, struct sock *);
 } sockets_per_process SEC(".maps");
 
+// src_dst_per_process keeps track of the src and dst information
+// between tcp_v4/6_connect and the return of inet_stream_connect
+// Since tcp_v4/6_connect is directly inside of inet_stream_connect,
+// we do not need to keep a large number of entries.
```

Comment:
Is it true with synchronous sockets? Could the thread block in the `tcp_v4_connect` function, and if we have more than max_entries=32 threads on the system blocked on that function, the map could get full.

Reply:
That is also true. What size would you suggest?

Reply:
For now I increased it from `32` to `128`